### PR TITLE
Fix windows build prob: yaml, prometheus, mbedtls, libuv

### PR DESCRIPTION
**Pull request by bingyanglin**
Fix part of #34

### CODE REVIEW

File: build/BUILD.civetweb [hunk 1]
```diff
@@ -18,7 +23,10 @@ cc_library(
         "-DNO_SSL",
         "-DNO_FILES",
     ],
-    linkopts = ["-lpthread"],
+    linkopts = select({
+        ":win": ["-lpthread", "-lWs2_32",],
```
Comment (oopsmonk) [inline build/BUILD.civetweb]: bazel has provided the windows config.
```suggestion
        "@bazel_tools//src/conditions:windows": ["-lpthread", "-lWs2_32",],
```

File: build/BUILD.civetweb [hunk 2]
```diff
@@ -1,5 +1,10 @@
 licenses(["notice"])  # MIT license
 
+config_setting(
```
Comment (oopsmonk) [inline build/BUILD.civetweb]: remove

File: build/BUILD.libuv [hunk 3]
```diff
@@ -131,6 +137,10 @@ cc_library(
     "@bazel_tools//src/conditions:windows": LIBUV_WINNT_HEADERS,
     "//conditions:default": LIBUV_UNIX_HEADERS + LIBUV_LINUX_HEADERS,
   }),
+  linkopts = select({
+    ":win": ["-lWs2_32", "-liphlpapi", "-luv", "-lmswsock", "-ladvapi32", "-lpsapi", "-luserenv",],
```
Comment (oopsmonk) [inline build/BUILD.libuv]:
```suggestion
    "@bazel_tools//src/conditions:windows": ["-lWs2_32", "-liphlpapi", "-luv", "-lmswsock", "-ladvapi32", "-lpsapi", "-luserenv",],
```

File: build/BUILD.libuv [hunk 4]
```diff
@@ -115,6 +115,12 @@ LIBUV_LINUX_HEADERS = [
   "include/uv/linux.h",
 ]
 
+config_setting(
```
Comment (oopsmonk) [inline build/BUILD.libuv]: remove

File: build/BUILD.mbedtls [hunk 5]
```diff
@@ -1,9 +1,18 @@
...
+    name = "win",
+    values = {"cpu": "x64_windows"},
+)
+
 cc_library(
     name = "mbedtls",
     srcs = glob(["library/*.c"]),
     hdrs = glob(["include/mbedtls/*.h"]),
+    linkopts = select({
+        ":win": ["-lwsock32", "-lWs2_32",],
```
Comment (oopsmonk) [inline build/BUILD.mbedtls]:
```suggestion
        "@bazel_tools//src/conditions:windows": ["-lwsock32", "-lWs2_32",],
```

File: build/BUILD.mbedtls [hunk 6]
```diff
@@ -1,9 +1,18 @@
 licenses(["notice"])  # Apache 2.0
 
+config_setting(
```
Comment (oopsmonk) [inline build/BUILD.mbedtls]: remove

File: build/BUILD.civetweb (hunk 1 above)
Comment (bingyanglin) [reply build/BUILD.civetweb]: Modified, thanks.

File: build/BUILD.civetweb (hunk 2 above)
Comment (bingyanglin) [reply build/BUILD.civetweb]: Modified, thanks.

File: build/BUILD.libuv (hunk 3 above)
Comment (bingyanglin) [reply build/BUILD.libuv]: Modified, thanks.

File: build/BUILD.libuv (hunk 4 above)
Comment (bingyanglin) [reply build/BUILD.libuv]: Modified, thanks.

File: build/BUILD.mbedtls (hunk 5 above)
Comment (bingyanglin) [reply build/BUILD.mbedtls]: Modified, thanks.

File: build/BUILD.mbedtls (hunk 6 above)
Comment (bingyanglin) [reply build/BUILD.mbedtls]: Modified, thanks.

File: build/BUILD.libuv [hunk 7]
```diff
@@ -131,6 +131,10 @@ cc_library(
     "@bazel_tools//src/conditions:windows": LIBUV_WINNT_HEADERS,
     "//conditions:default": LIBUV_UNIX_HEADERS + LIBUV_LINUX_HEADERS,
   }),
+  linkopts = select({
+    "@bazel_tools//src/conditions:windows": ["-lWs2_32", "-liphlpapi", "-luv", "-lmswsock", "-ladvapi32", "-lpsapi", "-luserenv",],
```
Comment (oopsmonk) [inline build/BUILD.libuv]: it's odd, loading libuv in libuv?
```suggestion
    "@bazel_tools//src/conditions:windows": ["-lWs2_32", "-liphlpapi", "-lmswsock", "-ladvapi32", "-lpsapi", "-luserenv",],
```

File: build/BUILD.libuv [hunk 8]
```diff
@@ -131,6 +131,10 @@ cc_library(
     "@bazel_tools//src/conditions:windows": LIBUV_WINNT_HEADERS,
     "//conditions:default": LIBUV_UNIX_HEADERS + LIBUV_LINUX_HEADERS,
   }),
+  linkopts = select({
+    "@bazel_tools//src/conditions:windows": ["-lWs2_32", "-liphlpapi", "-lmswsock", "-ladvapi32", "-lpsapi", "-luserenv",],
```
Comment (oopsmonk) [inline build/BUILD.libuv]: `pthread` and `dl` are needed on linux
```suggestion
    "@bazel_tools//src/conditions:windows": ["-lWs2_32", "-liphlpapi", "-lmswsock", "-ladvapi32", "-lpsapi", "-luserenv",],
    "@bazel_tools//src/conditions:linux_x86_64":["-pthread", "-ldl"],
```

Comment (bingyanglin) [reply build/BUILD.libuv]: Modified. Thank you for pointing this out.

**Pull request merged**